addons/medical_engine/functions/fnc_handleDamage.sqf

```diff
@@ -38,13 +38,64 @@ private _newDamage = _damage - _oldDamage;
-_unit setVariable [format [QGVAR($%1), _hitPoint], [_realDamage, _newDamage]];
-TRACE_3("Received hit",_hitpoint,_newDamage,_realDamage);
+TRACE_4("Received hit",_hitpoint,_ammo,_newDamage,_realDamage);
 
-// Engine damage to these hitpoints controls blood visuals, limping, weapon sway
-// Handled in fnc_damageBodyPart, persist here
-if (_hitPoint in ["hithead", "hitbody", "hithands", "hitlegs"]) exitWith {_oldDamage};
+// Drowning doesn't fire the EH for each hitpoint so the "ace_hdbracket" code never runs
+// Damage occurs in consistent increments
+if (
+    _hitPoint isEqualTo "#structural" &&
+    {getOxygenRemaining _unit <= 0.5} &&
+    {_damage isEqualTo (_oldDamage + 0.005)}
+) exitWith {
+    TRACE_5("Drowning",_unit,_shooter,_instigator,_damage,_newDamage);
+    [QEGVAR(medical,woundReceived), [_unit, [[_newDamage, "Body", _newDamage]], _unit, "drowning"]] call CBA_fnc_localEvent;
+
+    0
+};
+
+// Crashing a vehicle doesn't fire the EH for each hitpoint so the "ace_hdbracket" code never runs
+// It does fire the EH multiple times, but this seems to scale with the intensity of the crash
+private _vehicle = vehicle _unit;
+if (
+    EGVAR(medical,enableVehicleCrashes) &&
+    {_hitPoint isEqualTo "#structural"} &&
+    {_ammo isEqualTo ""} &&
+    {_vehicle != _unit} &&
+    {vectorMagnitude (velocity _vehicle) > 5}
+    // todo: no way to detect if stationary and another vehicle hits you
+) exitWith {
+    TRACE_5("Crash",_unit,_shooter,_instigator,_damage,_newDamage);
+    [QEGVAR(medical,woundReceived), [_unit, [[_newDamage, _hitPoint, _newDamage]], _unit, "vehiclecrash"]] call CBA_fnc_localEvent;
+
+    0
+};
+
+// Being inside an exploding vehicle doesn't trigger for each hitpoint
+// It seems to fire twice with ammo type "FuelExplosion" or "FuelExplosionBig"
+if (
+    _hitPoint isEqualTo "#structural" &&
+    {_damage == 1} &&
+    {_ammo isKindOf "FuelExplosion"} &&
+    {_vehicle != _unit}
+) exitwith {
+    // triggers twice, so do half damage each time. not very important as it's basically always lethal
+    private _hit = GET_NUMBER(configFile >> "CfgAmmo" >> _ammo >> "indirectHit", 10)/2;
+    private _uniform = uniform _unit;
+    if (_uniform isEqualTo "") then {
+        _uniform = getText (configOf _unit >> "nakedUniform");
+    };
+    private _uniformClass = GET_STRING(configFile >> "CfgWeapons" >> _uniform >> "ItemInfo" >> "uniformClass", "U_BasicBody");
+    private _damages = [];
+    {
+        private _armor = [_unit, _x] call FUNC(getHitpointArmor);
+        private _shielding = GET_NUMBER(configFile >> "CfgVehicles" >> _uniformClass >> "Hitpoints" >> _x >> "explosionShielding", 1);
+        _damages pushBack [_hit*_shielding/_armor, ALL_BODY_PARTS select _forEachIndex, _hit*_shielding]
+    } forEach ALL_HITPOINTS;
+    TRACE_6("Vehicle explosion",_unit,_shooter,_instigator,_damage,_newDamage,_damages);
+    [QEGVAR(medical,woundReceived), [_unit, _damages, _unit, _ammo]] call CBA_fnc_localEvent;
+
+    0
+};
```

Comment:
I've added this to manually calculate damage for people inside exploding vehicles, because I had a flash of inspiration while looking at the code. Might be out of scope for this PR so I can move it if desired.

Reply:
I'm assuming it's dependent on the other functional changes in the PR? If so I think it's fine here as it's not a huge addition to review at the same time, I'd just avoid any further scope creep.

If it's not I'd prefer it in a separate PR.

Reply:
The format will be affected by these changes, but no I'll remove it and revisit after this pr is sorted